extension/src/plots/model/collect.ts

```diff
@@ -487,18 +501,42 @@ export const collectTemplates = (data: PlotsOutput): TemplateAccumulator => {
   return acc
 }
 
-const fillTemplate = (template: string, datapoints: unknown[]) =>
-  JSON.parse(
-    template.replace('"<DVC_METRIC_DATA>"', JSON.stringify(datapoints))
+const fillTemplate = (
+  template: string,
+  datapoints: unknown[],
+  field?: string
+) => {
+  if (!field || !isConcatenatedField(field)) {
+    return JSON.parse(
+      template.replace('"<DVC_METRIC_DATA>"', JSON.stringify(datapoints))
+    ) as TopLevelSpec
+  }
+
+  const fields = unmergeConcatenatedFields(field)
+  return JSON.parse(
+    template.replace(
+      '"<DVC_METRIC_DATA>"',
+      JSON.stringify(
+        datapoints.map(data => {
+          const obj = data as Record<string, unknown>
+          return {
+            ...obj,
+            [field]: mergeFields(fields.map(field => obj[field] as string))
```

Comment:
[F] There is a situation where we concatenate fields together for the stoke dash update. If that happens we need to do the same thing with the data.